airbyte-config/models/src/main/resources/types/StandardSyncInput.yaml

```diff
@@ -35,11 +35,16 @@ properties:
   catalog:
     description: the configured airbyte catalog
     type: object
+    # necessary because the configuration declaration is in a separate package.
     existingJavaType: io.airbyte.protocol.models.ConfiguredAirbyteCatalog
   state:
     description: optional state of the previous run. this object is defined per integration.
     "$ref": State.yaml
   resourceRequirements:
-    description: optional resource requirements to run sync workers
+    description: optional resource requirements to run sync workers (confusing - overrides source and destination). takes precedence over source and destination resource requirements.
```

Comment:
Is there a reason why we don't just sweep this `resourceRequirements` field? Are there cases where we would actually rather define a single resourceRequirements that is applied to both the source and destination rather than just defining them separately?

Reply:
I suppose the answer to this is that our `connection` database table only has a single `resourceRequirements` column, which is where we can configure resource requirements for a single connection.

Being able to configure resource requirements for specific connections is definitely important, as I could see connections that are syncing a ton of data needing more memory/cpu than other connections that use the same connectors. It feels like it would be slightly more ideal to allow this connection-specific configuration to specify different resource requirements for the source and the destination connectors in the connection

Reply:
yeah. there is definitely a follow up to do here to rethink this.